[GTK][WPE] Undeprecate console message API and make it available in 2022 API

carlosgcampos opened the pull request:
#### e41ec2c29b8e53c5a5548c720195a808ac40124e
<pre>
[GTK][WPE] Undeprecate console message API and make it available in 2022 API
<a href="https://bugs.webkit.org/show_bug.cgi?id=275311">https://bugs.webkit.org/show_bug.cgi?id=275311</a>

Reviewed by Adrian Perez de Castro.

People have complained about this since we removed it, as long as we have the
web process extensions API there&apos;s no reason to not have this too.

* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformGTKDeprecated.cmake:
* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/PlatformWPEDeprecated.cmake:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesGTKDeprecated.txt:
* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/SourcesWPEDeprecated.txt:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitConsoleMessage.cpp:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitConsoleMessage.h.in:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitConsoleMessagePrivate.h:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebPage.cpp:
(webkitWebPageDidSendConsoleMessage):
(webkit_web_page_class_init):
(webkitWebPageCreate):
* Source/WebKit/WebProcess/InjectedBundle/API/glib/webkit-web-process-extension.h.in:
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestConsoleMessage.cpp:
(ConsoleMessageTest::~ConsoleMessageTest):
* Tools/TestWebKitAPI/Tests/WebKitGLib/WebProcessExtensionTest.cpp:
(consoleMessageSentCallback):
(pageCreatedCallback):
* Tools/TestWebKitAPI/glib/CMakeLists.txt:

Canonical link: <a href="https://commits.webkit.org/279899@main">https://commits.webkit.org/279899@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9483f83f130f0fbb38f386305b8164ded2286ba2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/54778 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/34210 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/7353 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/58056 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/5509 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/57078 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/41756 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/5523 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/44368 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/3727 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/56873 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/32338 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/47457 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/25492 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/29128 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/4794 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/3650 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/50945 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/5011 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/59646 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/30024 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/5155 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/51788 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/31166 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/47539 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/51199 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/32175 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8126 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/30954 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aperezdc commented on Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitConsoleMessage.cpp:
```diff
@@ -35,8 +33,6 @@ ALLOW_DEPRECATED_DECLARATIONS_END
  *
  * Since: 2.12
- *
- * Deprecated: 2.40
```
A bit of a pity that `gi-docgen` does not allow `Deprecated: 2.40 - 2.42` 😛

carlosgcampos committed:
[GTK][WPE] Undeprecate console message API and make it available in 2022 API
https://bugs.webkit.org/show_bug.cgi?id=275311

Reviewed by Adrian Perez de Castro.

People have complained about this since we removed it, as long as we have the
web process extensions API there's no reason to not have this too.

* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformGTKDeprecated.cmake:
* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/PlatformWPEDeprecated.cmake:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesGTKDeprecated.txt:
* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/SourcesWPEDeprecated.txt:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitConsoleMessage.cpp:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitConsoleMessage.h.in:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitConsoleMessagePrivate.h:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitWebPage.cpp:
(webkitWebPageDidSendConsoleMessage):
(webkit_web_page_class_init):
(webkitWebPageCreate):
* Source/WebKit/WebProcess/InjectedBundle/API/glib/webkit-web-process-extension.h.in:
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestConsoleMessage.cpp:
(ConsoleMessageTest::~ConsoleMessageTest):
* Tools/TestWebKitAPI/Tests/WebKitGLib/WebProcessExtensionTest.cpp:
(consoleMessageSentCallback):
(pageCreatedCallback):
* Tools/TestWebKitAPI/glib/CMakeLists.txt:

Canonical link: https://commits.webkit.org/279899@main
